### ISSUE TITLE
[Feature Request] A guide on how to (safely) reduce LAMMPS disk usage

### ISSUE THREAD

**Issue by arn-all** [lammps/lammps#2883]
**Summary**

The standard installation from source leads to a large disk usage (both the repo and built files).
Some users that are space-limited on computing clusters or have several independant LAMMPS installations may find it useful to know how they can gain space without affecting LAMMPS's behaviour. 

Some tips allow to dramatically reduce this disk usage and obtain some kind of "lightweight LAMMPS distribution", but informations are scattered in the docs pages and side effects are not totally clear to me.

**Detailed Description**

As discussed in issue (#1839), the repo is pretty big and becomes even larger after building the code.

From my reading here, I collected some tips that can be used:

- clone only the last commit instead of the full project history `git clone git@github.com:lammps/lammps --depth=1 --branch=master` (cuts by half the downloaded size but doesn't allow for browsing the different versions of lammps)
- [build LAMMPS without the debug](https://docs.lammps.org/Build_basics.html#including-or-removing-debug-support) `-g` flag, which saves a lot of space. If I understand well this will only affect the use of [debugging tools](https://docs.lammps.org/Errors_debug.html) (GDB, valgrind).
- The `liblammps_mpi.a` file is smaller in the `shared` build mode (#1977) I'm not sure to understand which would be the side effects of switching the mode if I'm only interested in the executable. Is it safe to delete this file after build ?
- Running `make clean-all` after build is a simple way of removing files generated during build that are no more necessary.
- Remove the examples, potentials, bench, unitttest and doc folders if they are not used.

Is there any other ways of reducing LAMMPS's size ? Do you think putting these together in a docs section would be relevant ?

**Comment by akohlmey** [comment]
@arn-all Yes, I think this would be a good item to add to the manual e.g. as `Notes on saving disk space` entry in a `Build_diskspace.rst` file added to the `Build.rst` TOC.

Some thoughts of my own on this:

- What and how you can save disk space depends on how you compile and how you download LAMMPS.
  - The tar files from lammps.org contain the prebuilt html documentation and thus 
  - When you compile with the traditional make based method, you have object files and library files scattered throughout the lib 
and src folders and only the Obj_* folders will be removed with "make clean-all" the LAMMPS executables and main library remains as well as all compiled content in the lib tree.
  - When you compile with CMake you must do an out-of-source build and this build folder does not have to be within the source tree. When I compile LAMMPS on external machines, the home folder is often on a (somewhat sluggish) networked file system, so I tend to create the CMake build folder in a local scratch file or ramdisk (often on /tmp) and then copy the resulting executable (and library) to where I need it or configure a proper install prefix (defaults to $HOME/.local) and then do `make install`. One can also place the build folder(s) on a scratch file system and thus avoid having it add to the filesystem quota. Building out-of-source has the added advantage that no file is created in the source folder and thus the used space can be easily regained by deleting the build folder(s)
  - Compiling without debug info can save diskspace, but I would recommend against it unless you don't expect and crashes or needs to debug code (or just get a stack trace for a bug report). Instead, I would recommend to use the strip command *after* the executable has been properly validated and deemed ready for production runs (this can also be applied to liblammps.so if a shared library was built).
  - When compiling a static LAMMPS library, the library (.a) file can indeed be deleted, if needed. When building with a shared library, both the executable and the shared library must be kept (but in that case the executable will be *much* smaller).
  - When using git clone the --depth flag can be used to reduce the size of the .git folder, but when downloading and unpacking a tar or zip file, that folder is not even present. It can also be deleted, if not git operations are intended. The .git folder will contain another copy of the entire distribution (in compressed format). The files that "blow up" the .git folder were added a long time ago, so even using --depth 1000 or so should not make a big difference and that will give you access or at least a couple of recent patch releases. the benefit of using git clone is that you can add feature branches with local changes and always switch between your customizations and the original or easily compare. If you don't plan on doing that, the .git folder can also simply be deleted.
  - other folders that are not needed to compile/use LAMMPS are "potentials" (in some cases), "examples", "bench", "doc", "tools" (in most cases), "unittest" (in most cases). A few very large potential files are only downloaded on demand when the corresponding package is used (that is currently an extra 120MByte)

As a point of reference, here is a current snapshot of a LAMMPS checkout that I have:
```
4	README
4	SECURITY.md
20	fortran
20	LICENSE
480	cmake
736	python
6388	unittest
18692	tools
23476	lib
35748	doc
37416	bench
42256	src
53840	potentials
173824	examples
571168	.git

203392	build-dielectric
227588	build-clang
440808	build-mingw
456364	build-ubuntu
528608	build-windows
618120	build-dpd
664156	build-opencl
1678892	build-pgi
2343736	build-most
2456740	build-rann
5044240	build-test
9706456	build-intel
```

Two things stand out: 
1) the large amount (multi-GB) of storage is only needed for compiling so doing an out-of-source compile with CMake in a scratch file system is a good way avoid wasting disk quota
2) outside of the .git and examples folders the savings potential on the existing folders is rather limited.

**Comment by akohlmey** [comment]
I forgot to mention, that when building with CMake you can compile multiple configurations (e.g. with different MPI libs or compilers or different features) from the same source code and when doing development you can save space by using the [git worktree command](https://git-scm.com/docs/git-worktree) to connect multiple git checkouts (e.g. to have different (feature) branches checked out at the same time) to the same .git folder.
